homeassistant/components/shelly/coordinator.py

```diff
@@ -536,6 +543,8 @@ async def _async_update_data(self) -> None:
             raise UpdateFailed(f"Device disconnected: {repr(err)}") from err
         except InvalidAuthError:
             self.entry.async_start_reauth(self.hass)
+        except FirmwareUnsupported as err:
```

Comment:
Can this be raised in `ShellyRestCoordinator` or `ShellyBlockCoordinator`?

Reply:
Actually nothing in the coordinator handles ConfigEntryNotReady

Reply:
https://github.com/home-assistant/core/blob/066a0ccc6d3d95a8b61087d2044f3a8b7244b780/homeassistant/helpers/update_coordinator.py#L361

We only handle ConfigEntryAuthFailed

Reply:
Oh didn't check helper, was concentrated into integration code :cry:

Reply:
It looks like sleeping devices will set last_error instead. Given we are going to switch to the design we talked about earlier (move the checks to HA) this will all get refactored anyways so let's take it out for now as we only need this repair issue until we do that.